AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt

```
@@ -0,0 +1,309 @@
+                                // do nothing
+                            }
+
+                            override fun onTabReselected(tab: TabLayout.Tab) {
+                                // do nothing
+                            }
+                        },
+                    )
+                }
+            }
+        }
+    }
+
+    val noteEditorWatcher: TextWatcher =
+        object : TextWatcher {
+            /**
+             * Declare a nullable variable refreshPreviewerFragmentRunnable of type Runnable.
+             * This will hold a reference to the Runnable that refreshes the previewer noteEditorFragment.
+             * It is used to manage delayed noteEditorFragment updates and can be null if no updates in card.
+             */
+            private var refreshPreviewerFragmentRunnable: Runnable? = null
+
+            override fun afterTextChanged(arg0: Editable) {
+                refreshPreviewerFragmentRunnable?.let { refreshPreviewerFragmentHandler.removeCallbacks(it) }
+                val updateRunnable =
+                    Runnable {
+                        loadNoteEditorPreviewer()
+                    }
+                refreshPreviewerFragmentRunnable = updateRunnable
+                refreshPreviewerFragmentHandler.postDelayed(updateRunnable, REFRESH_NOTE_EDITOR_PREVIEW_DELAY)
+            }
+
+            override fun beforeTextChanged(
+                arg0: CharSequence,
+                arg1: Int,
+                arg2: Int,
+                arg3: Int,
+            ) {
+                // do nothing
+            }
+
+            override fun onTextChanged(
+                arg0: CharSequence,
+                arg1: Int,
+                arg2: Int,
+                arg3: Int,
+            ) {
+                // do nothing
+            }
+        }
+
+    /**
+     * Retrieves the [NoteEditorFragment]
+     */
+    val noteEditorFragment: NoteEditorFragment?
```

This can return null but we're not handling it properly in other places.

Problem: When you call `noteEditorFragment?.someMethod()`, if fragment is not found, the code will just silently fail.